### ISSUE TITLE
Feature request: Reformat gather output to include taxonomy

### ISSUE THREAD

**Issue by brooksph** [dib-lab/sourmash#384]
Can we modify the gather output to separate matches into columns based on taxonomic rank? It's sometimes difficult to do this at the command line when the species or strain id spans multiple fields. For example, KQ235715.1 Fusobacterium nucleatum subsp. animalis D11 genomic scaffold adfWA-supercont2.1. I think @ctb may have already done this for lca.

**Comment by ctb** [comment]
note that we have no requirement for taxonomic information at present in the
gather database.  so have to think about how we could integrate it when
it is present but do without when it isn't.

**Comment by brooksph** [comment]
That's tricky but "undefined" might work. Not sure if there are standards but I'll look around.

**Comment by ctb** [comment]
Just revisiting this with some more thoughts -

gather works by finding the signature among the search subjects that best matches the hashes in the query, subtracting the matched hashes, and then repeating with the remaining hashes.  The name output by gather comes from the name of each found signature.  So to fix this we would have to update signatures to have taxonomic information (which is a big burden on the user - it's a reason I frequently don't use the lca search!)

But it might be possible to do the same 'gather' algorithm but with an lca database... so you'd have 'sourmash lca gather' that would output taxonomic info. Humm.

**Comment by ctb** [comment]
See #390; example output so far:

```
./sourmash lca gather ../lca-db//tully-genome-sigs/TOBG_IN-33.fna.gz.sig ../lca-db/delmont-MAGs-k31.lca.json.gz 
loaded 1 databases.
ksize=31 scaled=10000
0.8 Mbp    0.7%    Archaea; Euryarchaeota; ; ; ; ; 
110.0 kbp    0.1%    Eukaryota; Chlorophyta; Prasinophyceae; Mamiellales; Bathycoccaceae; Ostreococcus; 

25.8% (310.0 kbp) have no assignment.
```

**Comment by ctb** [comment]
Closed by #390.
